src/linear_algebra/matrix/pos_def.lean

```diff
@@ -20,14 +20,16 @@ quadratic forms.
 
 namespace matrix
 
-variables {R : Type*} [ordered_semiring R] [star_ring R] {n : Type*} [fintype n]
+variables {𝕜 : Type*} [is_R_or_C 𝕜] {n : Type*} [fintype n]
 
 open_locale matrix
 
 /-- A matrix `M : matrix n n R` is positive definite if it is hermitian
    and `xᴴMx` is greater than zero for all nonzero `x`. -/
-def pos_def (M : matrix n n R) :=
-M.is_hermitian ∧ ∀ x : n → R, x ≠ 0 → 0 < dot_product (star x) (M.mul_vec x)
+def pos_def (M : matrix n n 𝕜) :=
+M.is_hermitian ∧ ∀ x : n → 𝕜, x ≠ 0 → 0 < is_R_or_C.re (dot_product (star x) (M.mul_vec x))
```

Comment:
What was the motivation for this change? Isn't this strictly less general than before, as now we can't talk about matrices of integers being positive definite?

Reply:
The old definition `0 < dot_product (star x) (M.mul_vec x)` relies on the partial order on the complex numbers. But this order does not exist for `is_R_or_C`. (And if we added it, it would not be definitionally equal to the order on the reals...)

Reply:
Am I right in thinking there was a Zulip thread about this?

> And if we added it, it would not be definitionally equal to the order on the reals..

Not if we made it a field of `is_R_or_C`

Reply:
Yes, the Zulip thread is: https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.2314794.20inner.20prod.20of.20matrix

If we make the order a field of `is_R_or_C`, the order on C would become global.